### PR TITLE
Feature/index structure

### PR DESCRIPTION
**Pull request by Hodgy**
Enables admin users to build the structure of the Azure Search index for the site.

- Choose which fields for each post type should be indexed
- Choose which indexed fields are facetable ( Should only really be set for fields that contain fixed input )

### CODE REVIEW

File: includes/class-wp-lucene-search-fields.php [hunk 1]
```diff
@@ -121,11 +139,41 @@ public function get_core_fields()
...
+     *
+     * Will also pull out fields from post_object and repeater fields
+     *
+     * @param $post_type
+     * @return array
+     */
+    public function get_fields_for_post_type( $post_type )
+    {
+        $all_fields = [];
+        foreach ( $this->get_field_groups_for_post_type( $post_type ) as $field_group ) {
```
Comment (cooperaj) [inline includes/class-wp-lucene-search-fields.php]: Consistency in bracket spacing?

File: includes/class-wp-lucene-search-fields.php [hunk 2]
```diff
@@ -121,11 +139,41 @@ public function get_core_fields()
...
+            $new_fields = $this->get_fields_for_group( $field_group );
+            foreach ($new_fields as $new_field) {
```
Comment (cooperaj) [inline includes/class-wp-lucene-search-fields.php]: Consistency in bracket spacing?

File: includes/class-wp-lucene-search-fields.php (hunk 1 above)
Comment (Hodgy) [reply includes/class-wp-lucene-search-fields.php]: Fixed

File: includes/class-wp-lucene-search-fields.php (hunk 2 above)
Comment (Hodgy) [reply includes/class-wp-lucene-search-fields.php]: Fixed

**Pull request merged**